app/filtering.js

```diff
@@ -577,10 +577,6 @@ function registerPermissionHandler (session, partition) {
       } else if (isFullscreen && alwaysAllowFullscreen) {
         // Always allow fullscreen if setting is ON
         response.push(true)
-      } else if (isOpenExternal && (
-        // The Brave extension and PDFJS are always allowed to open files in an external app
-        isPDFOrigin || isBraveOrigin)) {
```

Comment:
I think this can actually be fixed by checking against the requestingUrl instead of the "mainFrameURL". I don't think the url of the webcontents is ever safe to use for any of these checks and Chrome also uses the requestingUrl in all but one case.

Reply:
@bridiver in this case the requesting URL is 'mailto:...' or whatever the external protocol is, so it would never be a PDF origin or Brave origin